internal/server/grpc/grpc.go

```diff
@@ -258,33 +275,99 @@ func loadX509KeyPair(certfile, keyfile string) (tls.Certificate, error) {
 	return tls.X509KeyPair(cert, key)
 }
 
-func generateCertificatesForATLS() ([]byte, []byte, error) {
+func generateCertificatesForATLS(caUrl string, cvmId string) ([]byte, []byte, error) {
 	curve := elliptic.P256()
+
 	privateKey, err := ecdsa.GenerateKey(curve, rand.Reader)
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to generate private/public key: %w", err)
 	}
 
-	certTemplate := &x509.Certificate{
-		SerialNumber: big.NewInt(202403311),
-		Subject: pkix.Name{
+	var certDERBytes []byte
+
+	if caUrl == "" || cvmId == "" {
```

Comment:
currently for client (cli) we disable tls verification when atls is enabled since cert is generated by agent. However with this change we can allow provision of ca cert to cli and in this case we don't have to skip tls verification. so enable tls verification for atls when ca is provided and load the ca cert on client ca pool https://github.com/ultravioletrs/cocos/blob/ebe119686fe9c13faa99ca8cc9c293987d9758e2/pkg/clients/grpc/atls.go#L30